### PR TITLE
[SPARK-40133][SQL][TESTS] Update excluded TPC-DS queries golden files

### PR DESCRIPTION
**Pull request by wangyum**
### What changes were proposed in this pull request?
1. Move the `regenerateGoldenFiles` parameter to `SparkFunSuite`.
2. Update [excluded TPC-DS queries](https://github.com/apache/spark/blob/559fe96a4082aad70bda262c820ca3cfe888c151/sql/core/src/test/scala/org/apache/spark/sql/TPCDSBase.scala#L41) golden files if `regenerateGoldenFiles` is true.

### Why are the changes needed?

Track changes to these query plans.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

N/A.

### CODE REVIEW

**Review by HyukjinKwon — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

cc @cloud-fan

**Review by dongjoon-hyun — COMMENTED** [review]
BTW, SPARK-40124 landed to the master, https://github.com/apache/spark/commit/3153f1e3799344609a17ec9458ba6f4d7e7ecb82 .

Do you think you can rebase this to the master branch once more?

File: core/src/test/scala/org/apache/spark/SparkThrowableSuite.scala [hunk 1]
```diff
@@ -45,7 +45,6 @@ class SparkThrowableSuite extends SparkFunSuite {
         "core/testOnly *SparkThrowableSuite -- -t \"Error classes are correctly formatted\""
    }}}
    */
-  private val regenerateGoldenFiles: Boolean = System.getenv("SPARK_GENERATE_GOLDEN_FILES") == "1"
```
Comment (cloud-fan) [inline core/src/test/scala/org/apache/spark/SparkThrowableSuite.scala:48]: sorry I may miss some context. What problem do we fix by moving this to `SparkFunSuite`?

Comment (wangyum) [reply core/src/test/scala/org/apache/spark/SparkThrowableSuite.scala:48]: `TPCDSBase` needs this parameter:
image

**Review by cloud-fan — APPROVED** [review]
good catch!

**Comment by wangyum** [comment]
Thank you all. Merged to master.

**Pull request closed without merging**